front/lib/api/files/upsert.ts

```diff
@@ -309,85 +304,58 @@ type ProcessingFunction = ({
   dataSource: DataSourceResource;
 }) => Promise<Result<undefined, Error>>;
 
-type ProcessingPerUseCase = {
-  [k in FileUseCase]: ProcessingFunction | undefined;
-};
+const getProcessingFunction = ({
+  contentType,
+  useCase,
+}: {
+  contentType: SupportedFileContentType;
+  useCase: FileUseCase;
+}): ProcessingFunction | undefined => {
+  // Use isSupportedDelimitedTextContentType() everywhere to have a common source of truth
+  if (isSupportedDelimitedTextContentType(contentType)) {
```

Comment:
Similar approach to the `upload.ts` file but with the usage of `isSupportedDelimitedTextContentType()` that we also use to determine what is "table-queryable" with JIT actions.